### PR TITLE
General indexer implementation

### PR DESCRIPTION
**Pull request by ars9**

### CODE REVIEW

File: client.rs/src/client/implementations/evm.rs [hunk 1]
```diff
@@ -47,9 +47,16 @@ impl HapiCoreEvm {
...
             .map_err(|e| ClientError::UrlParseError(format!("`provider-url`: {e}")))?;
 
-        let signer = LocalWallet::from_str(options.private_key.unwrap_or_default().as_str())
-            .map_err(|e| ClientError::Ethers(format!("`private-key`: {e}")))?
-            .with_chain_id(options.chain_id.unwrap_or(31337_u64));
+        let signer = LocalWallet::from_str(
+            options
+                .private_key
+                .unwrap_or(
+                    "0000000000000000000000000000000000000000000000000000000000000001".to_string(),
```
Comment (ProstoSergei) [inline client.rs/src/client/implementations/evm.rs:54]: Why this string is used?

File: client.rs/tests/assert.rs [hunk 2]
```diff
@@ -30,11 +30,15 @@ macro_rules! assert_tx_output {
...
 
-        value
+        let tx = value
             .get("tx")
             .and_then(serde_json::Value::as_str)
             .map(|s| s.to_owned())
-            .expect("`tx` key not found or not a string")
+            .expect("`tx` key not found or not a string");
+
+        println!("tx: {}", tx);
```
[the remark this answers is not in the document]
Comment (ars9) [reply client.rs/tests/assert.rs:39]: It is a debug log that can be viewed in nocapture mode when running tests. I've needed it a couple of times already.

File: client.rs/src/client/implementations/evm.rs (hunk 1 above)
Comment (ars9) [reply client.rs/src/client/implementations/evm.rs:54]: We can't initialize a client without specifiying a wallet. We can't initialize a wallet without specifyng a private key. This private key is sufficiently self-descriptive about the fact that it's not an actual secret.

**Comment by hlgltvnnk** [comment]
root/indexer/src/indexer/state - line 39 - "If the new state is processing, and the current state is also processing, just move on" - this line is redundant, because state is unchanging

**Comment by ars9** [comment]
> (This is not part of MR) `root/indexer/src/indexer/state - line 39` - "If the new state is processing, and the current state is also processing, just move on" - this line is redundant, because state is unchanging

It is not redundant, because this state change can update cursor, because the state equality is checked without it.

**Pull request merged**